.azure-pipelines.yml

```diff
@@ -155,8 +155,15 @@ jobs:
   - bash: |
       # set options
       set -o nounset errexit pipefail
-      # run tests (exclude pytest.AMReX when running Python tests)
-      ctest --test-dir build --output-on-failure -E AMReX
+      # determine if the build was triggered by a push to the development branch
+      if [[ "$(Build.SourceBranch)" == "refs/heads/development" ]]; then
+        # run tests (exclude pytest.AMReX when running Python tests)
+        # and submit results to CDash as Continuous
+        ctest --test-dir build --output-on-failure -E AMReX -D Continuous
+      else
+        # run tests (exclude pytest.AMReX when running Python tests)
+        ctest --test-dir build --output-on-failure -E AMReX
+      fi
```

Comment:
This is what I would try first, based on the information reported in https://github.com/ECP-WarpX/WarpX/pull/5566#issuecomment-2596485383.

Reply:
Again on this, and on the issues reported in https://github.com/ECP-WarpX/WarpX/pull/5566#issuecomment-2594256505, reading the documentation more carefully it seems to me that the various "steps" (e.g., `Start`, `Update`, `Configure`, `Build`, `Test`, `MemoryCheck`, `Coverage`, `Submit`) can be run independently only for the `Nightly` or `Experimental` modes, but not for the other modes such as `Continuous`. This could explain what we observed so far.

![Screenshot from 2025-01-17 11-09-18](https://github.com/user-attachments/assets/3c2474c9-2fc8-415f-812c-cf783b7e2c89)

Reply:
However, `Continuous` still seems like the right choice for main branch events, such as merging PRs into the main branch.

Reply:
```suggestion
      # determine if the build was triggered by a push to the development branch
      if [[ "$(Build.SourceBranch)" == "refs/heads/development" ]]; then
        # run tests (exclude pytest.AMReX when running Python tests)
        # and submit results to CDash as Experimental
        ctest --test-dir build --output-on-failure -E AMReX \
          -D ExperimentalTest -D ExperimentalSubmit
      else
        # run tests (exclude pytest.AMReX when running Python tests)
        ctest --test-dir build --output-on-failure -E AMReX
      fi
```